src/content/1.7/development/multistore/configuration-forms/_index.md

```diff
@@ -0,0 +1,138 @@
+
+```php
+<?php
+class MyFormType extends TranslatorAwareType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add(
+                'my_field',
+                SwitchType::class,
+                [
+                    'required' => true,
+                    // Here we add the multistore_configuration_key attribute with the configuration key it is linked to.
+                    'attr' => [
+                        'multistore_configuration_key' => 'THE_CONFIGURATION_KEY',
+                    ],
+                    // ...
+                ]
+            )
+        // ...
+
+```
+
+Last but not least, you must make the frontend aware of these state modifications, you just need to add the `MultistoreConfigField` in your javascript:
+
+```js
+window.prestashop.component.initComponents(
+    [
+        'MultistoreConfigField',
+    ],
+);
+````
+
+
+
+Doing this will have several effects:
+
+- Each field implementing the `multistore_configuration_key` will have a multistore checkbox before them ;
+- Checkboxes will automatically be checked or not depending on the configuration for the current shop context ;
+- When the configuration value is inherited, the checkbox will be unchecked, and the field will be disabled and greyed out ;
+- Thanks to the javascript component, checking or unchecking a checkbox will enable/disable the field for the current shop context.
+
+
+## Saving data from a multistore configuration form
+
+When receiving the data from the form, you will also receive values from the multistore checkboxes that were added to your form:
+
+- When a multistore checkbox is pushed unchecked, we want to delete the configuration value for the current context (so that it inherits its value from parent contexts).
+- When a multistore checkbox is pushed checked, we want to save the configuration value for the current context.
+
+When your configuration form is multistore compatible, your configuration class should extend the `AbstractMultistoreConfiguration` abstract class instead of directly implementing the `DataConfigurationInterface` interface. This will give you access to helper methods that will make it easy to store the right configuration values at the right place and for the right context.
+
+The two helper methods are
+
+- **`getShopConstraint()`:** this will give you a ShopConstraint object reflecting the current context
+```php
+public function getShopConstraint(): ?ShopConstraint
+```
+- **`updateConfigurationValue()`:** this will save or delete the configuration value for the given configuration key, taking into account the state of the multistore checkbox and the current shop context. Note that this helper method is totally compatible with non multistore fields.
```

Comment:
```suggestion
- **`updateConfigurationValue()`:** this will save or delete the configuration value for the given configuration key, and is aware of the multistore checkbox and the current shop context. Note that this helper method is fully compatible with not-multistore fields.
```